node/node_test.go

```diff
@@ -789,35 +789,122 @@ func TestIntervalDownlink(t *testing.T) {
 				UseLittleEndian: tt.useLittleEndian, Header: tt.header, TestOnly: !tt.testGatewayReturn,
 			}
 			resp, err := testNode.SendIntervalDownlink(ctx, req)
-			if tt.expectedErr != "" {
-				test.That(t, resp, test.ShouldBeNil)
-				test.That(t, err.Error(), test.ShouldContainSubstring, tt.expectedErr)
-			} else {
-				test.That(t, err, test.ShouldBeNil)
-
-				// receive a response from the gateway
-				if tt.testGatewayReturn {
-					// we should receive a success from the gateway
-					gatewayResp, gatewayOk := resp[GatewaySendDownlinkKey].(string)
-					test.That(t, gatewayOk, test.ShouldBeTrue)
-					test.That(t, gatewayResp, test.ShouldEqual, tt.expectedReturn)
-
-					// we should not receive a interval success message
-					nodeResp, nodeOk := resp[IntervalKey].(string)
-					test.That(t, nodeOk, test.ShouldBeFalse)
-					test.That(t, nodeResp, test.ShouldEqual, "")
-				} else {
-					// we should not receive a success from the gateway
-					gatewayResp, gatewayOk := resp[GatewaySendDownlinkKey].(string)
-					test.That(t, gatewayOk, test.ShouldBeFalse)
-					test.That(t, gatewayResp, test.ShouldEqual, "")
-
-					// we should receive a interval payload message
-					nodeResp, nodeOk := resp[IntervalKey].(string)
-					test.That(t, nodeOk, test.ShouldBeTrue)
-					test.That(t, nodeResp, test.ShouldEqual, tt.expectedReturn)
-				}
+			testDoCommandResp(t, resp, err, IntervalKey, tt.expectedReturn, tt.expectedErr, tt.testGatewayReturn)
+		})
+	}
+}
+
+func TestResetDownlink(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	deps, tmpDir := testutils.NewNodeTestEnv(t, gatewayNames, nodeNames, "decoder.js")
+	// copy the path to the tmpDir
+	testDecoderPath := fmt.Sprintf("%s/%s", tmpDir, "decoder.js")
+
+	validConf := resource.Config{
+		Name: testNodeName,
+		ConvertedAttributes: &Config{
+			Decoder:  testDecoderPath,
+			Interval: &testInterval,
+			JoinType: JoinTypeOTAA,
+			DevEUI:   testDevEUI,
+			AppKey:   testAppKey,
+		},
+	}
+
+	n, err := newNode(ctx, deps, validConf, logger)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, n, test.ShouldNotBeNil)
+	testNode := n.(*Node)
+
+	tests := []struct {
+		name              string
+		header            string
+		payload           string
+		expectedReturn    string
+		testGatewayReturn bool
+		expectedErr       string
+	}{
+		{
+			name:              "valid reset",
+			header:            "FF",
+			payload:           "003C",
+			expectedReturn:    "FF003C",
```

Comment:
Do you want to add a lowercase payload to ensure your capitalization logic is maintained in the future?

Reply:
the lowercase test uses a lowercase payload and header